drivers/sensor/ccs811/ccs811.c

```diff
@@ -0,0 +1,230 @@
+		val->val2 = uval % 1000000;
+
+		break;
+	case SENSOR_CHAN_VOC:
+		uval = drv_data->voc * 1000000;
+		val->val1 = uval / 1000000;
+		val->val2 = uval % 1000000;
+
+		break;
+	case SENSOR_CHAN_VOLTAGE:
+		/*
+		 * Voltage readings are contained in least
+		 * significant 10 bits in volts
+		 */
+		uval = (drv_data->resistance & CCS811_VOLTAGE_MASK)
+					* CCS811_VOLTAGE_SCALE;
+		val->val1 = uval / 1000000;
+		val->val2 = uval % 1000000;
+
+		break;
+	case SENSOR_CHAN_CURRENT:
+		/*
+		 * Current readings are contained in most
+		 * significant 6 bits in microAmps
+		 */
+		uval = drv_data->resistance >> 10;
+		val->val1 = uval / 1000000;
+		val->val2 = uval % 1000000;
+
+		break;
+	default:
+		return -ENOTSUP;
+	}
+
+	return 0;
+}
+
+static const struct sensor_driver_api ccs811_driver_api = {
+	.sample_fetch = ccs811_sample_fetch,
+	.channel_get = ccs811_channel_get,
+};
+
+static int switch_to_app_mode(struct device *i2c)
+{
+	u8_t status, buf;
+
+	SYS_LOG_DBG("Switching to Application mode...");
+
+	if (i2c_reg_read_byte(i2c, CONFIG_CCS811_I2C_ADDR,
+			      CCS811_REG_STATUS, &status) < 0) {
+		SYS_LOG_ERR("Failed to read Status register");
+		return -EIO;
+	}
+
+	/* Check for the application firmware */
+	if (!(status & CCS811_STATUS_APP_VALID)) {
+		SYS_LOG_ERR("No Application firmware loaded");
+		return -EINVAL;
+	}
+
+	buf = CCS811_REG_APP_START;
+	/* Set the device to application mode */
+	if (i2c_write(i2c, &buf, 1, CONFIG_CCS811_I2C_ADDR) < 0) {
+		SYS_LOG_ERR("Failed to set Application mode");
+		return -EIO;
+	}
+
+	if (i2c_reg_read_byte(i2c, CONFIG_CCS811_I2C_ADDR,
+			      CCS811_REG_STATUS, &status) < 0) {
+		SYS_LOG_ERR("Failed to read Status register");
+		return -EIO;
+	}
+
+	/* Check for application mode */
+	if (!(status & CCS811_STATUS_FW_MODE)) {
+		SYS_LOG_ERR("Failed to start Application firmware");
+		return -EINVAL;
+	}
+
+	SYS_LOG_DBG("CCS811 Application firmware started!");
+
+	return 0;
+}
+
+int ccs811_init(struct device *dev)
+{
+	struct ccs811_data *drv_data = dev->driver_data;
+	int ret;
+	u8_t hw_id, status;
+
+	drv_data->i2c = device_get_binding(CONFIG_CCS811_I2C_MASTER_DEV_NAME);
+	if (drv_data->i2c == NULL) {
+		SYS_LOG_ERR("Failed to get pointer to %s device!",
+			    CONFIG_CCS811_I2C_MASTER_DEV_NAME);
+		return -EINVAL;
+	}
+
+	/*
+	 * Wakeup pin should be pulled low before initiating any I2C transfer.
+	 * If it has been tied to GND by default, skip this part.
+	 */
+#ifdef CONFIG_CCS811_GPIO_WAKEUP
+	drv_data->gpio = device_get_binding(CONFIG_CCS811_GPIO_DEV_NAME);
+	if (drv_data->gpio == NULL) {
+		SYS_LOG_ERR("Failed to get pointer to %s device!",
+			    CONFIG_CCS811_GPIO_DEV_NAME);
+		return -EINVAL;
+	}
+
+	gpio_pin_configure(drv_data->gpio, CONFIG_CCS811_GPIO_WAKEUP_PIN_NUM,
+			   GPIO_DIR_OUT);
+	gpio_pin_write(drv_data->gpio, CONFIG_CCS811_GPIO_WAKEUP_PIN_NUM, 0);
+
+	k_sleep(1);
+#endif
+
+	/* Switch device to application mode */
+	ret = switch_to_app_mode(drv_data->i2c);
+	if (ret)
```

Comment:
I missed this one. Please add brackets.

Reply:
Ah... Fixed this one as well as one more. Thanks for pointing out!